__tests__/app.test.js

```diff
@@ -0,0 +1,164 @@
+const db = require("../db/connection.js");
+
+const testData = require("../db/data/test-data");
+const app = require("../app.js");
+const seed = require("../db/seeds/seed.js");
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("/api/topics", () => {
+  describe("GET", () => {
+    test("200 - responds with array of topic objects with properties: slug, description", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((res) => {
+          expect(res.body.topics).not.toEqual([]);
+          expect(res.body.topics.length).toBe(3);
+          res.body.topics.forEach((topic) => {
+            expect.objectContaining({
+              slug: expect.any(String),
+              description: expect.any(String),
+            });
+          });
+        });
+    });
+  });
+  test("404 - given non existent path responds with message path not found <GLOBAL>", () => {
+    return request(app)
+      .get("/api/not-a-path")
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("path not found");
+      });
+  });
+});
+
+//ticket 14
+describe("/api/articles/:article_id", () => {
+  describe("GET", () => {
+    test("200 - responds with article object with properties: author<username from users table>, title, article_id, body, topic, created_at, votes", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then((res) => {
+          expect(res.body.article).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: expect.any(String),
+            votes: 100,
+          });
+        });
+    });
+    test("400 - given invalid id data type, returns message: bad request", () => {
+      return request(app)
+        .get("/api/articles/not-an-id")
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("bad request");
+        });
+    });
+    test("404 - given non existent id with no data, returns message: not found", () => {
+      return request(app)
+        .get("/api/articles/999999")
+        .expect(404)
+        .then((res) => {
+          expect(res.body.msg).toBe("not found");
+        });
+    });
+  });
+
+  //ticket 7
+  describe("PATCH", () => {
+    test("200 - accepts body in form of { inc_votes: newVote } with newVote dictating the inc/decrement of votes. Responds with updated article", () => {
+      const votes = { inc_votes: -10 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(votes)
+        .expect(200)
+        .then((res) => {
+          expect(res.body.article).toEqual({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
```

Comment:
Should be an easy one - but have you tested if it actually increments the votes?
At the moment, all the tests check that is decrements.